include/RAJA/util/zip.hpp

```diff
@@ -57,7 +57,7 @@ struct ZipIterator
   }
 
   template < typename... Args,
-             typename = concepts::enable_if<DefineConcept(concepts::convertible_to<Iters>(camp::val<Args&&>()))...> >
+             typename = concepts::enable_if<type_traits::convertible_to<Args&&, Iters>...> >
```

Comment:
I don't think you want the r-value reference (&&) as part of the type in the check for `convertible_to`.

The pack expansion should also be within an `all_of` (I believe camp has a metafunction for truth-like types)

Reply:
The intent here is to have a general constructor that is only enabled if it can forward its arguments to the constructor for the tuple of iterators. I was trying to keep the actual reference type of the argument lvalue or rvalue. That doesn't matter as much here because this makes copies of the iterators. However this does matter in zip_tuple which can store references.

I think concepts::enable_if does the all_of for me.

Reply:
Shoot, you’re right about the enable_if ... silly me is used to std::enable_if

Regarding the && ... that makes sense. Sorry!